src/v/cluster/rm_stm.cc

```diff
@@ -1784,8 +1784,12 @@ void rm_stm::apply_fence(model::record_batch&& b) {
 
     auto [fence_it, _] = _log_state.fence_pid_epoch.try_emplace(
       bid.pid.get_id(), bid.pid.get_epoch());
-    if (fence_it->second < bid.pid.get_epoch()) {
+    // using less-or-equal to update tx_seqs on every transaction
+    if (fence_it->second <= bid.pid.get_epoch()) {
```

Comment:
do we want to change this logic for both fence batch versions ?

Reply:
right, it doesn't matter if old version does it too